### PR TITLE
weldr/upload: don't expose AWS/Azure acconut details

### LINKED ISSUE #907: AWS credentials for previous builds available via API
After starting a compose with AWS upload enabled, the AWS credentials are available to anyone via the API:

```
# composer-cli --json compose info 3ada5fd7-7eee-4623-bee3-cb462493a6d2
{
    -- SNIP --
    "uploads": [
        {
            "uuid": "68f2d2e6-81ba-4fe7-a39a-a7d8047a9e9f",
            "status": "RUNNING",
            "provider_name": "aws",
            "image_name": "major-aws-test",
            "creation_time": 1596460680.0325832,
            "settings": {
                "region": "us-east-2",
                "accessKeyID": "SUPERSECRETACCESSKEY😱",
                "secretAccessKey": "SUPERSECRETACCESSKEY😱",
                "bucket": "...",
                "key": ""
            }
        }
    ]
}
```

Should we make these available via the API or hide them in some way?

### PR DESCRIPTION
**Pull request by teg**
Change the translation from our internal structs to the structs used for
weldr serialization to drop account details. These must obviously be
passed in to configure an upload, but exposing them in the logs may be
surprising.

There is no notion of user accounts in the weldr API, and the state
should not be considered private. However, this is likely to take people
by surprise, so let us guard the secrets entrusted to us.

Fixes #907.

Signed-off-by: Tom Gundersen <teg@jklm.no>

### CODE REVIEW

File: internal/weldr/upload.go [hunk 1]
```diff
@@ -112,19 +115,17 @@ func targetsToUploadResponses(targets []*target.Target, state common.ComposeStat
...
 			}
 			uploads = append(uploads, upload)
 		case *target.AzureTargetOptions:
 			upload.ProviderName = "azure"
 			upload.Settings = &azureUploadSettings{
-				StorageAccount:   options.StorageAccount,
-				StorageAccessKey: options.StorageAccessKey,
-				Container:        options.Container,
+				Container: options.Container,
+				// StorageAccount and StorageAccessKey is intentionally not included.
```
Comment (ondrejbudai) [inline internal/weldr/upload.go:128]:
```suggestion
				// StorageAccount and StorageAccessKey are intentionally not included.
```

File: internal/weldr/upload.go [hunk 2]
```diff
@@ -88,6 +88,9 @@ func (u *uploadRequest) UnmarshalJSON(data []byte) error {
 // This ignore the status in `targets`, because that's never set correctly.
 // Instead, it sets each target's status to the ImageBuildState equivalent of
 // `state`.
+//
+// This also ignores any sensitive data passed into targets. Access keys may
+// be passed as input to composer, but sholud not be possibley to queried.
```
Comment (ondrejbudai) [inline internal/weldr/upload.go:93]:
```suggestion
// be passed as input to composer, but should not be possible to be queried.
```

File: internal/weldr/upload.go (hunk 1 above)
Comment (teg) [reply internal/weldr/upload.go:128]: Hehe, as you can tell I first intended to only exclude the one ;)

**Comment by ondrejbudai** [comment]
The openstack failure is clearly unrelated.

**Pull request merged**